File: Beginner/Roman_to_Int/roman_to_int.cpp
// Converting Roman to Integer
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Solution {
    public:
        int romanToInt(std::string s) {
            int ans = 0;

            std::vector<int> roman(128);
            roman['I'] = 1;
            roman['V'] = 5;
            roman['X'] = 10;
            roman['L'] = 50;
            roman['C'] = 100;
            roman['D'] = 500;
            roman['M'] = 1000;

            for (int i = 0; i + 1 < s.length(); ++i) {
                if (roman[s[i]] < roman[s[i+1]]) { ans -= roman[s[i]]; }
                else { ans += roman[s[i]]; }
            }
            return ans + roman[s.back()];

        }
};

int main() {
    Solution s;
    std::cout << "Enter the Roman number: ";
    std::string roman;
    std::cin >> roman;
    std::cout << s.romanToInt(roman) << std::endl;
    return 0;
}

File: Beginner/Roman_to_Int/roman_to_int.py
class Solution:
    def romanToInt(self,  s:str) -> int:
        roman = {
            "I":1,
            "V":5,
            "X":10,
            "L":50,
            "C":100,
            "D":500,
            "M":1000
        }

        N = len(s)
        i = N-1
        output = 0;
        while i >= 0:
            if i < N-1 and roman[s[i]] < roman[s[i+1]]:
                output -= roman[s[i]]
            else:
                output += roman[s[i]]
            i -= 1
        return output

num = input("Please enter the roman numeral: ")
s = Solution()
print(f"The number is: {s.romanToInt(num)}");
